### DISCUSSION TITLE
Include ZX to also add scripting capability

### DISCUSSION THREAD

**Discussion by yerlaser** [oven-sh/bun#688] [Ideas]
Hi,
Would like to suggest to bundle https://github.com/google/zx
Motivation: this will elevate 'bun' to a very powerful scripting host.
I envision that `import zx` would trigger inclusion of all relevant modules.
This will allow the same executable `bun` to handle MJS scripts with the following shebang with all the 'zx' convenience:

`#!/usr/bin/env bun`

**Comment by Jarred-Sumner** [comment]
Once `child_process` is implemented, you should be able to use `zx` the same as Node

**Reply by yerlaser** [reply]
Hi @Jarred-Sumner,

Thanks for the response!
Just to be clear, my suggestion is to include ZX inside BUN by default.
In this case, without any external dependancies BUN can be used for build scripts, or even management scripts.

**Reply by Jarred-Sumner** [reply]
That's interesting. I do think it would be worth having a cross-platform shell built-in to `bun run`. That shell could be exposed to JS this way, too.

**Reply by yerlaser** [reply]
Great! Thanks for considering!
